Libmesh update (#21127)
idaholab/moose: modules/doc/content/newsletter/2022_05.md

  compatibility with some compilers and compiler warning settings, and
  support for `std::isinf()`.

### `2022.05.24` Update

- Even more internal smart pointers.  This includes changes to the
  `Parameters` class that may require downstream changes to match;
  compatibility in Moose itself has already been fixed.
- When trying to combine a low-order `Elem` with an incompatibly

>   higher order `FE` type, the mistake is now always reported with
>   comprehensible error message, even in non-debugging-mode builds.

Reviewer: ```suggestion
  higher order `FE` type, the mistake is now always reported with a
  comprehensible error message, even in non-debug-mode builds.
```

- When a parsed function is used and the parser returns an error, we

>   now check and report details even in non-debugging-mode builds.

Reviewer: ```suggestion
  now check and report details even in non-debug-mode builds.
```

- The last use of C++17-deprecated std::iterator has been fixed;
  a few compilers outside CI were giving us warnings about it.
- We put support for input meshes that themselves have holes into

>   MeshedHole, so you can mesh with holes while your hole has holes.

Reviewer: ```suggestion
  `MeshedHole`, so you can mesh with holes while your hole has holes.
```

- Bug fixes for some triangulator use cases.
- Compatibility fixes for some `PETSc` 3.17 configurations.

> 

Reviewer: ```suggestion
```

- MetaPhysicL updates, for evaluating `pow(negative DualNumber,
  integer)` operations without throwing floating point exceptions,
  and also evaluating binary functions faster in general when mixing
  DualNumber and non-DualNumber parameter types.

> 

Reviewer: ```suggestion
```
